crates/rolldown/src/utils/chunk/render_chunk_exports.rs

```diff
@@ -321,9 +317,6 @@ pub fn get_export_items(
 
   match chunk.kind {
     ChunkKind::EntryPoint { module, is_user_defined, .. } => {
-      if options.preserve_modules && !is_user_defined {
```

Comment:
This hack code is not needed since https://github.com/rolldown/rolldown/pull/4626/files